src/lib/api/contrib/invenio/InvenioSearchApi.js

```diff
@@ -78,9 +81,18 @@ export class InvenioSearchApi {
    * @param {string} stateQuery the `query` state with the user input
    */
   async search(stateQuery) {
-    const response = await this.http.request({
+    let response = await this.http.request({
       params: stateQuery,
     });
-    return this.responseSerializer.serialize(response.data);
+    response = this.responseSerializer.serialize(response.data);
+    const newQueryState = updateQueryState(
+      stateQuery,
+      response.extras,
+      STORE_KEYS
+    );
+    if (!_isEmpty(newQueryState)) {
+      response.newQueryState = newQueryState;
+    }
+    return response;
```

Comment:
delete the `extras` before? (double check if it works if not set!)
```suggestion
    delete response.extras
    return response;
```